dashboard/modules/reporter/reporter_head.py

```diff
@@ -31,6 +32,20 @@ def __init__(self, dashboard_head):
         self._stubs = {}
         self._ray_config = None
         DataSource.agents.signal.append(self._update_stubs)
+        # TODO(fyrestone): Avoid using ray.state in dashboard, it's not
+        # asynchronous and will lead to low performance. ray disconnect()
+        # will be hang when the ray.state is connected and the GCS is exit.
+        # Please refer to: https://github.com/ray-project/ray/issues/16328
+        assert dashboard_head.gcs_address or dashboard_head.redis_address
```

Comment:
Reporter agent is the place to aggregate metrics, so I added the prometheus writing logic into reporter head